### ISSUE TITLE
Improve color scheming

### ISSUE THREAD

**Issue by Silzinc** [RPG-Alex/project-julia#2]
The color scheming I had when testing the product was hardly readable, in my opinion. 

Moreover, it did not take advantage of the number of iterations evolving as a gradient as we get far from the Julia set. This is a specificity of Julia and Mandelbrot sets that allow a very nice continuous coloring.

The idea would be to modify the `julia` function to make it return a floating point between 0 and 1, and use this point to choose a color in a gradient. I used this definition for a color gradient example:

```rs
use bevy::prelude::*;
type Rgba = UVec4;

const DEFAULT_RGBA_COLORS: [Rgba; 8] = [
  UVec4::new(0, 0, 0, 255),
  UVec4::new(4, 35, 59, 255),
  UVec4::new(37, 131, 204, 255),
  UVec4::new(226, 232, 142, 255),
  UVec4::new(209, 128, 42, 255),
  UVec4::new(214, 15, 15, 255),
  UVec4::new(140, 6, 6, 255),
  UVec4::new(0, 0, 0, 255),
];

const DEFAULT_TRESHOLDS: [f32; 8] = [0.03, 0.05, 0.08, 0.1, 0.13, 0.18, 0.25, 1.];

#[derive(Resource)]
pub struct ColorGradient<const N: usize>
{
  colors:    [Rgba; N],
  tresholds: [f32; N],
}

pub const DEFAULT_COLOR_GRADIENT: ColorGradient<8> = ColorGradient {
  colors:    DEFAULT_RGBA_COLORS,
  tresholds: DEFAULT_TRESHOLDS,
};

impl Default for ColorGradient<8>
{
  fn default() -> Self { DEFAULT_COLOR_GRADIENT }
}

impl<const N: usize> ColorGradient<N>
{
  #[inline]
  pub fn new(colors: [Rgba; N], tresholds: [f32; N]) -> Self { Self { colors, tresholds } }

  #[inline]
  pub fn get_color(&self, x: f32) -> Rgba
  {
    if x <= self.tresholds[0] {
      return self.colors[0];
    }
    for i in 0..N {
      if self.tresholds[i] > x {
        let nx = (self.tresholds[i] - x) / (self.tresholds[i] - self.tresholds[i - 1]);
        let float_color = self.colors[i].as_vec4() * (1.0 - nx) + self.colors[i - 1].as_vec4() * nx;
        let color = float_color.as_uvec4();
        return color;
      }
    }
    *self.colors.last().unwrap()
  }
}
```
More generally, it would be great and handy to replace the computations on [0, 255] integers by computations on [0, 1] floating points, which would make us replace `UVec4` by `Color` in the snippet above, as well as `(u8, u8, u8, u8)` by `Color`. Coloring a pixel in `map_value_to_color` becomes very simple then:
```rs
fn map_value_to_color(value: f32) -> (u8, u8, u8, u8)
{
  let color = DEFAULT_COLOR_GRADIENT.get_color(value).to_array();
  (color[0], color[1], color[2], color[3])
}
```
Or, using `Color` everywhere
```rs
fn map_value_to_color(value: f32) -> Color
{
  DEFAULT_COLOR_GRADIENT.get_color(value)
}
```

**Comment by RPG-Alex** [comment]
I think that is a very clever way to do it. I would also be curious to see the performance gains of doing this. I will assign this to you as it seems like a good place to start optimizing. For now I won't branch and we can stay on main, but just for now.

**Comment by Silzinc** [comment]
After some attempts, the easiest appears indeed to use `bevy::prelude::Color` almost everywhere. I also tried `bevy::prelude::Vec4` but stuck with `Color`, mostly because it comes with an API to manipulate different color representations and defines certain basic colors (see [the doc](https://docs.rs/bevy/latest/bevy/prelude/enum.Color.html)).

The only place where `Vec4` could have been better was when I tried to reduce aliasing: the basic idea for this is, for each pixel, to compute the color for various points randomly chosen in the pixel, and take the mean of the results. We can sum colors, but not multiply them by a scalar or anything like that. This results in a little bit more verbose and less efficient code, but nothing much.
